### ISSUE TITLE
Sponsored Product Purchase Products Report Export vs API Output

### ISSUE THREAD

**Issue by tspicer** [amzn/ads-advanced-tools-docs#68]
Hello,

Using the new SP v3 Reporting API, we have observed variations between the output from the API and the report exported from Reporting. For example, the `spPurchasedProduct` API report type compared to the reporting UI export (i.e., Sponsored Product - Purchase Product - Daily).

If a report is exported from reporting, say for March 30th, 2023, and a request is made to the API for `spPurchasedProduct`, with the same date (DAILY), it will yield some overlap in records, but there will be rows not found in the API output and rows from the API not in the export. 

Typically, the report export, while not perfect, is usually a reasonable cross-reference to the output from the API, all other things being equal. 

In reviewing the docs, I had not seen anything that may explain variations between the two. Any clarification can help us better document these observations for our customers.

**Comment by elizafarley** [comment]
Hi @tspicer, thanks for this feedback. Small differences between the advertising console reports and the API can sometimes occur. We are planning to develop more robust content around differences between UI and API reporting for V3 (similar to [this documentation for V2](https://advertising.amazon.com/API/docs/en-us/reporting/v2/advertising-console)). I'll let you know here when this is available.

For now, if you are seeing significant differences between the advertising console and API reports, I'd suggest opening a [JIRA ticket](https://amzn-clicks.atlassian.net/servicedesk/customer/portal/2/group/2) so the support team can investigate whether or not this is bug.

**Comment by tspicer** [comment]
@elizafarley .
Thanks for the response. We will open a JIRA ticket as the delta between the two is 20-30% in pure row counts.

**Comment by elizafarley** [comment]
To clarify, is SB report generation failing for this advertiser or are you getting a blank report (empty json array)?

**Comment by tspicer** [comment]
Hi @elizafarley  we are getting a blank report (empty json array). We get the blank report, regardless of the day we request.

**Comment by elizafarley** [comment]
@tspicer Does your report request include `"creativeType": "all"`? If you leave this out, campaigns created using SB version 4 (including any campaigns created after September 2022 using the advertising console) may not be included in the report, resulting in an empty report.

**Comment by tspicer** [comment]
This is great information @elizafarley . We will check with the customer. Don't forget to add your answer to the FAQs at some point!

**Comment by tspicer** [comment]
@elizafarley , I'm not sure if `creativeType` will exist in the in the new v3 reporting API,  but I would suggest recommending the default be set to `all`. At the moment, the docs give the direction that  _If you are running both video and non-video Sponsored Brands campaigns, make sure to create two report requests: one for video and one for non-video._

Based on your suggestion, with v2, `all` should be used unless someone someone wants a `video` only report.

**Comment by elizafarley** [comment]
@tspicer Yes, I agree this FAQ is confusing given the behavior for SB campaigns created using version 4. I'll get that updated. The best practice is definitely to use `"creativeType": "all"` as the default. The team is working to streamline this in version 3 reporting, so I'll pass your feedback on.

**Comment by tspicer** [comment]
@elizafarley , one comment on  using `"creativeType": "all"` is that this is really an attribute of a video report. This attribute reflects a video report that includes non-video campaigns. Why? The results of using `"creativeType": "all"` does not support metrics found in non-video reports. For example, units sold are not available `"creativeType": "all".` If certain metrics like units sold are important, then using `"creativeType": "all"` will exclude those from the outputs. 

Reading `"creativeType": "all"` implies it  is a base report attribute, meaning a request with `all` for any report type will include a superset of data. Anything other than `all` acts as a filter of that superset according to the defined dimension (ie video). That is not the case.

The request logic in v2 is more complicated than it needs to be, with unpredicatable or counter intuitive outcomes based what I described.

**Comment by elizafarley** [comment]
@tspicer thanks for that feedback. This complexity has resolved from the introduction of the [Sponsored Brands version 4 campaign management API](https://advertising.amazon.com/API/docs/en-us/sponsored-brands/v4-migration-guide), which allows a single Sponsored Brands campaign to have both video and non-video creatives. Since in version 4 there is no distinction between video and non-video at the campaign level, only the metrics that work for video are available when using `"creativeType": "all"`.

This is something that the team is mindful of in the design of version 3 reporting for Sponsored Brands.

**Comment by hhwangg123** [comment]
@tspicer , I tool another look at your communication and the new features introduced by version 4. I believe that the issue is resolved, so I am closing this issues. Please let me know if you have any further questions. Thanks.
